### PR TITLE
[47057] Non-working Days/Holidays selection with 12.5 update

### PR DESCRIPTION
**Pull request by bsatarnejad**
https://community.openproject.org/work_packages/47057/activity

### CODE REVIEW

File: frontend/src/app/features/calendar/op-calendar.service.ts [hunk 1]
```diff
@@ -34,11 +34,13 @@ export class OpCalendarService extends UntilDestroyedMixin {
   }
 
   applyNonWorkingDay({ date }:{ date?:Date }, nonWorkingDays:IDay[]):string[] {
-    const formatted = moment(date).format('YYYY-MM-DD');
-    if (date && (this.weekdayService.isNonWorkingDay(date) || nonWorkingDays.find((el) => el.date === formatted))) {
-      return ['fc-non-working-day'];
+    if (date) {
+      const utcDate = new Date(date.getTime() + date.getTimezoneOffset() * 60000);
```
Comment (b12f) [inline frontend/src/app/features/calendar/op-calendar.service.ts]: Could you make the magic number more clear, perhaps explain in a comment or the commit why this is needed

File: frontend/src/app/shared/components/datepicker/modal-single-date-picker/modal-single-date-picker.component.ts [hunk 2]
```diff
@@ -270,8 +270,10 @@ export class OpModalSingleDatePickerComponent implements ControlValueAccessor, O
   }
 
   writeWorkingValue(value:string):void {
+    const date = new Date(value);
+    const utc = new Date(date.getTime() + date.getTimezoneOffset() * 60000);
```
Comment (b12f) [inline frontend/src/app/shared/components/datepicker/modal-single-date-picker/modal-single-date-picker.component.ts]: Same goes here of course :) I guess doing it in a squash commit is best

**Review by b12f — APPROVED** [review]
LGTM

**Comment by bsatarnejad** [comment]
If the user's system time zone is negative when selecting a date in single date-picker modal, it highlights the previous day instead of the selected date and it is due to new Date() function that uses local time, but after this fix we are using UTC Date to create a date object for the selected date.

**Pull request merged**